src/int/bswap.rs

```diff
@@ -0,0 +1,32 @@
+use crate::int::Int;
+
+fn bswap<I: Int + From<u8>>(x: I) -> I {
+    let mut ret = I::ZERO;
+    for (i, byte) in x.to_be_bytes().into_iter().enumerate() {
+        ret |= I::from(byte) << (i as u32 * 8);
+    }
+    ret
+}
+
+intrinsics! {
+    #[maybe_use_optimized_c_shim]
+    #[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
```

Comment:
One last question, why is this gated? I don't notice similar gating in https://github.com/llvm/llvm-project/blob/919b1ecafc010379eff88368b050068223a01f99/compiler-rt/lib/builtins/bswapsi2.c#L15

If it's related to AVR having a custom ABI for a lot of intrinsics, you can just mark these `#[avr_skip]` instead

Otherwise you should probably just `#![cfg(...)]`-gate the whole file since `bswap` will get an unused warning on 16-bit platforms.

Reply:
Yeah, I was worried about 16-bit targets, whichever they might be. You are right, they are not gated in compiler-rt, and they are specifically 32- and 64- bit, so should probably ungate them.

Reply:
Ungated si and di versions, left 128-bit under conditional though. I doubt it is ever available or called in 16-bit anything... Is there even any 128 bit int C type in such implementations?